apollo-entity/src/main/java/com/spotify/apollo/entity/CodecEntityMiddleware.java

```diff
@@ -19,79 +19,87 @@
  */
 package com.spotify.apollo.entity;
 
+import static java.util.concurrent.CompletableFuture.completedFuture;
+
 import com.spotify.apollo.RequestContext;
 import com.spotify.apollo.Response;
 import com.spotify.apollo.Status;
 import com.spotify.apollo.route.AsyncHandler;
 import com.spotify.apollo.route.Middleware;
 import com.spotify.apollo.route.SyncHandler;
-
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 import java.io.IOException;
 import java.util.Objects;
 import java.util.Optional;
 import java.util.function.Function;
-
 import javaslang.control.Either;
+import javax.annotation.Nullable;
 import okio.ByteString;
-
-import static java.util.concurrent.CompletableFuture.completedFuture;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * A factory for creating middlewares that can be used to create routes that deal directly
  * with an api entity.
  *
- * A {@link EntityCodec} is used to define how to go between a {@link ByteString} and the
- * entity type used in your route handlers.
+ * A {@link Codec} is used to define how to go between a {@link ByteString} and the entity type
+ * used in your route handlers.
  */
 class CodecEntityMiddleware implements EntityMiddleware {
 
   private static final Logger LOG = LoggerFactory.getLogger(CodecEntityMiddleware.class);
 
   private static final String CONTENT_TYPE = "Content-Type";
 
-  private final EntityCodec codec;
+  private final Codec codec;
+
+  // This is kept for backwards compatibility, will override value in EncodedResponse#contentType
+  // todo: remove in 2.0
+  @Nullable
   private final String contentType;
 
-  CodecEntityMiddleware(EntityCodec codec) {
-    this(codec, codec.defaultContentType());
+  CodecEntityMiddleware(Codec codec) {
+    this.codec = Objects.requireNonNull(codec);
+    this.contentType = null;
   }
 
-  CodecEntityMiddleware(EntityCodec codec, String contentType) {
+  @Deprecated
```

Comment:
Maybe add a pointer to a description for the rationale for deprecation, and what should be used instead.